src/ddprof_module_lib.cc

```diff
@@ -15,12 +15,33 @@
 #include <fcntl.h>
 #include <gelf.h>
 #include <libelf.h>
-#include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 
 namespace ddprof {
 
+class FileDescriptorHolder {
+public:
+  FileDescriptorHolder() : _fd(-1) {}
+  DDRes open_file(std::string_view path) {
+    _fd = ::open(path.data(), O_RDONLY);
+    if (_fd < 0) {
+      LG_WRN("[Mod] Couldn't open fd to module (%s)", path.data());
+      return ddres_warn(DD_WHAT_MODULE);
+    }
+    LG_DBG("[Mod] Success opening %s, ", path.data());
+    return ddres_init();
+  }
+
+  void take_ownership() { _fd = -1; }
+
+  ~FileDescriptorHolder() {
+    if (_fd > 0) {
```

Comment:
Indeed, closing 1, 2, 3 does not make sense. The class could be more generic, though here it is mostly a small wrapper to avoid thinking about exit paths.
The `open` API does return a negative number in case of error. The check was tailored to the file's logic.
We should consider this if we ever want to extract this class.
Also I should probably have put the class in an anonymous namespace.